src/alloc/isolated_alloc.rs

```diff
@@ -171,8 +168,20 @@ impl IsolatedAlloc {
 
     /// Expands the available memory pool by adding one page.
     fn add_page(&mut self) -> (*mut u8, &mut DenseBitSet<usize>) {
-        // SAFETY: The system page size, which is the layout size, cannot be 0
-        let page_ptr = unsafe { alloc::alloc(self.page_layout()) };
+        // SAFETY: mmap is safe to call as long as the length is a multiple of
+        // the system page size
```

Comment:
Where do the docs say that the size needs to be a multiple of the page size?

It seems to me like the call is always valid. The system will just give us more memory than we asked for if we request half a page.